lingpy/compare/partial.py

```diff
@@ -0,0 +1,544 @@
+    previously morphologically segmented. 
+
+    """
+
+    def __init__(self, infile, **keywords):
+
+        kw = {
+                "morphemes" : "morphemes",
+                "partial_cognates" : "partial_cognate_sets"
+                }
+        kw.update(keywords)
+        lingpy.compare.lexstat.LexStat.__init__(self, infile, **kw)
+        self._morphemes = kw['morphemes']
+        self._partials = kw['partial_cognates']
+
+    def _get_partial_matrices(
+            self,
+            concept=False,
+            method='sca',
+            scale=0.5,
+            factor=0.3,
+            restricted_chars='_T',
+            mode='global',
+            gop=-2,
+            restriction='',
+            **keywords
+            ):
+        """
+        Function creates matrices for the purpose of partial cognate detection.
+        """
+
+        # set the defaults
+        kw = dict(
+            defaults=False,
+            external_scorer=False,  # external scoring function
+            imap_mode= False,
+            separators=lingpy.settings.rcParams['morpheme_separator']+'+_#'
+        )
+        kw.update(keywords)
+
+        if method == 'lexstat':
+            # define the function with help of lambda
+            function = lambda idxA, idxB, sA, sB: calign.align_pair(
+                    self[idxA, self._numbers][sA[0]:sA[1]],
+                    self[idxB, self._numbers][sB[0]:sB[1]],
+                    [self.cscorer[_charstring(self[idxB, self._langid]), n]
+                        for n in self[idxA, self._numbers][sA[0]:sA[1]]],
+                    [self.cscorer[_charstring(self[idxA, self._langid]), n]
+                        for n in self[idxB, self._numbers][sB[0]:sB[1]]],
+                    self[idxA, self._prostrings][sA[0]:sA[1]],
+                    self[idxB, self._prostrings][sB[0]:sB[1]],
+                    1,
+                    scale,
+                    factor,
+                    self.cscorer,
+                    mode,
+                    restricted_chars,
+                    1)[2]
+        elif method == 'sca':
+            # define the function with help of lambda
+            function = lambda idxA, idxB, sA, sB: calign.align_pair(
+                    [n.split('.', 1)[1] for n in self[idxA,
+                        self._numbers][sA[0]:sA[1]]],
+                    [n.split('.', 1)[1] for n in self[idxB,
+                        self._numbers][sB[0]:sB[1]]],
+                        self[idxA, self._weights][sA[0]:sA[1]],
+                        self[idxB, self._weights][sB[0]:sB[1]],
+                        self[idxA, self._prostrings][sA[0]:sA[1]],
+                        self[idxB, self._prostrings][sB[0]:sB[1]],
+                    gop,
+                    scale,
+                    factor,
+                    self.rscorer,
+                    mode,
+                    restricted_chars,
+                    1)[2]
+
```

Comment:
Maybe a better alternative (assigning `lambda`s to a variable is considered bad style):
```python
def function(idxA, idxB, sA, sB):
    if method == 'lexstat':
        args = [...]
    elif method == 'sca':
        args = [...]
    else:
        raise ValueError
    return calign.align_pair(*args)[2]
```